hive/src/main/java/org/elasticsearch/hadoop/hive/pushdown/parse/EsTreeParser.java

```diff
@@ -0,0 +1,272 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.hadoop.hive.pushdown.parse;
+
+import org.elasticsearch.hadoop.hive.pushdown.Pair;
+import org.elasticsearch.hadoop.hive.pushdown.SargableParser;
+import org.elasticsearch.hadoop.hive.pushdown.TreeParser;
+import org.elasticsearch.hadoop.hive.pushdown.node.ConstantNode;
+import org.elasticsearch.hadoop.hive.pushdown.node.FieldNode;
+import org.elasticsearch.hadoop.hive.pushdown.node.Node;
+import org.elasticsearch.hadoop.hive.pushdown.node.OpNode;
+import org.elasticsearch.hadoop.hive.pushdown.parse.query.AndJson;
+import org.elasticsearch.hadoop.hive.pushdown.parse.query.BoolJson;
+import org.elasticsearch.hadoop.hive.pushdown.parse.query.ExistsJson;
+import org.elasticsearch.hadoop.hive.pushdown.parse.query.JsonObj;
+import org.elasticsearch.hadoop.hive.pushdown.parse.query.JsonObjManager;
+import org.elasticsearch.hadoop.hive.pushdown.parse.query.MatchJson;
+import org.elasticsearch.hadoop.hive.pushdown.parse.query.MissingJson;
+import org.elasticsearch.hadoop.hive.pushdown.parse.query.QueryJson;
+import org.elasticsearch.hadoop.hive.pushdown.parse.query.RangeJson;
+import org.elasticsearch.hadoop.hive.pushdown.parse.query.TermJson;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Set;
+
+public class EsTreeParser extends TreeParser<JsonObj> {
+    JsonObj preFilterJson = null;
+    List<JsonObj> queries = new LinkedList<JsonObj>();
```

Comment:
This variable never seems to be updated with any contents?

Reply:
del yet